packages/docsearch-react/src/utils/removeHighlightTags.ts

```diff
@@ -1,22 +1,23 @@
 import { DocSearchHit, InternalDocSearchHit } from './../types';
+import { getAttributeValueByPath } from './getAttributeValueByPath';
 
 const regexHighlightTags = /(<mark>|<\/mark>)/g;
 const regexHasHighlightTags = RegExp(regexHighlightTags.source);
 
 export function removeHighlightTags(
   hit: DocSearchHit | InternalDocSearchHit
 ): string {
-  if (
-    !(hit as InternalDocSearchHit).__docsearch_parent &&
-    !hit._highlightResult
-  ) {
+  const internalDocSearchHit = hit as InternalDocSearchHit;
+
+  if (!internalDocSearchHit.__docsearch_parent && !hit._highlightResult) {
     return hit.hierarchy.lvl0;
   }
 
-  const { value } = hit._highlightResult
-    ? hit._highlightResult.hierarchy.lvl0
-    : (hit as InternalDocSearchHit).__docsearch_parent!._highlightResult
-        .hierarchy.lvl0;
+  const value: string =
+    getAttributeValueByPath(
+      hit._highlightResult ? hit : internalDocSearchHit.__docsearch_parent,
+      ['_highlightResult', 'hierarchy', 'lvl0', 'value']
```

Comment:
could this be done with ?. as well to keep type safety?

Reply:
I've tried and thought it was a bit messy, so I went with what we use in Autocomplete. Would you prefer the other solution?

Reply:
isn't the use case in autocomplete with a user-provided path? in this case the path is static, so it's more efficient to do ?.. Why did you find it more messy?

Reply:
It's indeed not that messier than the other version, is that what you had in mind? https://github.com/algolia/docsearch/pull/1039/commits/2673afcd1aec63021840be1195a8a9d9ae5920d8